### ISSUE TITLE
coqc -quick should take a (documented) flag that informs the user of each task saved for later

### ISSUE THREAD

**Issue by Zimmi48** [Zimmi48/bugzilla-test#4125]
Note: the issue was created automatically with bugzilla2github tool

# Bugzilla Bug ID: 4125
Date: 2015-03-12 22:59:35 +0100
From: @_JasonGross
To:   
CC:   coq-bugs-redist@lists.gforge.inria.fr

Last updated: 2015-03-15 18:37:41 +0100

## Bugzilla Comment ID: 9845
Date: 2015-03-12 22:59:35 +0100
From: @_JasonGross

This would be very, very useful for making things like bug #4124 less surprising.  The rules for when a proof is processed asynchronously are neither simple nor free of bugs (e.g., they depend on the existence of .aux files, and, at present, depend on whether the lemma is defined at top-level or not, etc.), and [Set Suggest Proof Using] is insufficient to inform the user when there are proofs that "should" be parallelizable but are not (see bug #4124).

Thus, there should be a flag that tells the user what coq is *actually* doing, with regard to what proofs are deferred vs. not.  (Currently, determining this requires sticking a "fail" at the top of each proof assumed to be run in parallel and seeing which ones fail.)

## Bugzilla Comment ID: 9849
Date: 2015-03-12 23:27:48 +0100
From: @_gares

Well, if you pass -debug you get things like:
  
  master:0] Collecting proof ending at 12
  ..
  master:0] Synchronous 13 NoHint

Not exactly easy to read (12 refers to a system state, and you need graphviz installed + inspect a .pdf in /tmp to know which proof sends at 12), but isn't that what you are asking for?  Would it work for you if I print the lemma name near the second line up these?  Like

   proof: foo processed: Synchronous because: NoProofUsing_and_NoAuxFile

?

## Bugzilla Comment ID: 9851
Date: 2015-03-12 23:35:31 +0100
From: @_JasonGross

>    proof: foo processed: Synchronous because: NoProofUsing_and_NoAuxFile

Yes, if I can run `coqc -quick -debug foo 2>&1 | grep 'Synchronous because'` to see what Coq is doing, that would be useful.  It might also be nice to have 'Asynchronous because', so I can easily make sure I'm not missing any theorems I expect to be there.

I'm ... not sure if this should be documented.  It seems like it at least deserves a footnote in the user manual, that advanced users wanting to see which things are done asynchronously can look for such-and-such a label in the output of -debug.

Thanks!

## Bugzilla Comment ID: 9871
Date: 2015-03-13 17:18:17 +0100
From: @_gares

*** Bug #4128 has been marked as a duplicate of this bug. ***

## Bugzilla Comment ID: 9894
Date: 2015-03-15 18:37:41 +0100
From: @_gares

Will push soon
